crates/jxl-render/src/features/upsampling.rs

```diff
@@ -117,7 +117,14 @@ fn upsample_inner<const K: usize, const NW: usize>(
                     max = max.max(sample);
                 }
             }
-            grid_buf[y * frame_width + x] = sum.clamp(min, max);
+
+            grid_buf[y * frame_width + x] = if !(min <= max) {
```

Comment:
Clippy had an interesting lint...

```suggestion
            grid_buf[y * frame_width + x] = if min.partial_cmp(&max).is_none() {
```